test/lib/Elastica/Test/Query/ImageTest.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Elastica\Test\Query;
+
+use Elastica\Index;
+use Elastica\Query\Image;
+use Elastica\Test\Base as BaseTest;
+use Elastica\Type;
+
+class ImageTest extends BaseTest
+{
+    public function testToArrayFromReference()
+    {
+        $client = $this->_getClient();
+        $index = new Index($client, 'test');
+        $type = new Type($index, 'helloworld');
+        $field = "image";
+
+        $query = new Image();
+        $query->setFieldFeature($field, 'CEDD');
+        $query->setFieldHash($field, 'BIT_SAMPLING');
+        $query->setFieldBoost($field, 100);
+
+        $query->setImageByReference($field, $index->getName(), $type->getName(), 10);
+
+        $jsonString = '{"image":{"image":{"feature":"CEDD","hash":"BIT_SAMPLING","boost":100,"index":"test","type":"helloworld","id":10,"path":"image"}}}';
+        $this->assertEquals($jsonString, json_encode($query->toArray()));
+    }
+
+    public function testToArrayFromImage()
+    {
+        $field = "image";
+
+        $query = new Image();
+        $query->setFieldFeature($field, 'CEDD');
+        $query->setFieldHash($field, 'BIT_SAMPLING');
+        $query->setFieldBoost($field, 100);
+
+        $query->setFieldImage($field, __DIR__ . '/../../../../data/test.jpg');
```

Comment:
To have a more readable path, check here: https://github.com/ruflin/Elastica/blob/master/test/lib/Elastica/Test/IndexTest.php#L124